python/ray/monitor.py

```diff
@@ -106,25 +106,23 @@ def xray_heartbeat_batch_handler(self, unused_channel, data):
 
         message = ray.gcs_utils.HeartbeatBatchTableData.FromString(
             heartbeat_data)
-
         for heartbeat_message in message.batch:
-            num_resources = len(heartbeat_message.resources_available_label)
-            static_resources = {}
-            dynamic_resources = {}
-            for i in range(num_resources):
-                dyn = heartbeat_message.resources_available_label[i]
-                static = heartbeat_message.resources_total_label[i]
-                dynamic_resources[dyn] = (
-                    heartbeat_message.resources_available_capacity[i])
-                static_resources[static] = (
-                    heartbeat_message.resources_total_capacity[i])
+            total_resources = dict(
+                zip(heartbeat_message.resources_total_label,
+                    heartbeat_message.resources_total_capacity))
+            avail_resources = dict(
+                zip(heartbeat_message.resources_available_label,
+                    heartbeat_message.resources_available_capacity))
+            for res in total_resources:
+                avail_resources.setdefault(res, 0.0)
 
             # Update the load metrics for this raylet.
             client_id = ray.utils.binary_to_hex(heartbeat_message.client_id)
             ip = self.raylet_id_to_ip_map.get(client_id)
+            load_metrics_id = ip + "-" + client_id
```

Comment:
Is this the best way to do this?

Reply:
I think this needs a corresponding change to somewhere else. I currently see the following in my autoscaler logs:
> 2019-07-16 17:22:38,247 INFO autoscaler.py:198 -- LoadMetrics: Removed 1 stale ip mappings: {'10.138.0.90-e7a33a624c198a4a3280ed3ee9fbe3fdd13d287e'} not in {'10.138.0.94', '10.138.0.95', '10.138.0.91', '10.138.0.92', '10.138.0.90', '10.138.0.93'}

Reply:
Hm I think this might be wrong (or requires a larger change to get right)...